### PR TITLE
fixes/improvements to new error handling

### PR DESCRIPTION
**Pull request by moloko**
* fixes https://github.com/adaptlearning/adapt_framework/issues/2945
* improve the display/wording of the error messages
* allow the entirety of the SCORM error popup to be amended/translated 
* allow the error messages to be amended via the AAT (similar to how the 'player options' are set for the media component')
* add '(SCORM)' to the spoor extension's label in course configuration to make it clearer what it's for!
* trim very long (>80 chars) error message values as these screw up the window.confirm display in Firefox
* ensure empty values (e.g. when bookmarking to menu) are displayed as `''` in error messages
* remove the `trim` function, it was only needed for IE8

### CODE REVIEW

**Comment by moloko** [comment]
a few screenshots to show the changes...

The new, non-dismissable, 'could not connect to LMS' error, translated into French
image

**Comment by moloko** [comment]
Error setting suspend data (in Firefox):
image

**Comment by moloko** [comment]
Error getting lesson_location (IE11):
image

**Comment by moloko** [comment]
Error setting lesson_location on going to menu (IE11):
image

**Comment by moloko** [comment]
Finally, a commit error (Chrome):
image

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by oliverfoster — APPROVED** [review]

File: js/scorm/wrapper.js [hunk 1]
```diff
@@ -492,6 +492,13 @@ define([
         return;
       }
 
+      if ('value' in error.data) {
+        // because some browsers (e.g. Firefox) don't like displaying very long strings in the window.confirm dialog
+        if (error.data.value.length && error.data.value.length > 80) error.data.value = error.data.value.slice(0, 80) + '...';
```
Comment (tomgreenfield) [inline js/scorm/wrapper.js:497]: Just to point out the magic number of `80` and obviously doesn't take into account if someone's adjusted the error message. Probably best solved by switching to Notify and passing `value` into a read-only, scrollable text area.

**Review by tomgreenfield — APPROVED** [review]

File: js/scorm/error.js [hunk 2]
```diff
@@ -12,24 +12,25 @@ define(function() {
...
-    CLIENT_STATUS_UNSUPPORTED: `ScormWrapper::setStatus: the status '{{status}}' is not supported.`,
-    CLIENT_COULD_NOT_COMMIT: 'Course could not commit data to the LMS\nError {{code}}: {{info}}\nLMS Error Info: {{diagnosticInfo}}',
-    CLIENT_NOT_CONNECTED: 'Course is not connected to the LMS',
-    CLIENT_COULD_NOT_FINISH: 'Course could not finish',
-    CLIENT_COULD_NOT_GET_PROPERTY: 'Course could not get {{property}}\nError Info: {{info}}\nLMS Error Info: {{diagnosticInfo}}',
-    CLIENT_COULD_NOT_SET_PROPERTY: 'Course could not set {{property}} to {{value}}\nError Info: {{info}}\nLMS Error Info: {{diagnosticInfo}}',
-    CLIENT_INVALID_CHOICE_VALUE: 'Numeric choice/matching response elements must use a value from 0 to 35 in SCORM 1.2: {{value}}'
+    title: 'An error has occurred',
+    pressOk: `Press 'OK' to view detailed debug information to send to technical support.`,
+    CLIENT_COULD_NOT_CONNECT: 'The course could not connect to the Learning Management System',
```
[the remark this answers is not in the document]
Comment (moloko) [reply js/scorm/error.js:25]: > Could we switch to using Notify throughout in the future?

Yes, makes sense.

File: js/scorm/wrapper.js (hunk 1 above)
Comment (moloko) [reply js/scorm/wrapper.js:497]: > Just to point out the magic number of `80` and obviously doesn't take into account if someone's adjusted the error message. Probably best solved by switching to Notify and passing `value` into a read-only, scrollable text area.
> 
> See [something similar](https://github.com/adaptlearning/adapt_authoring/blob/910073836a0278488e8e384fa5a88cfbe1a6debe/routes/lang/en.json#L301) in the authoring tool.

Yep, I'm not entirely happy with this but feel it's 'good enough' for the moment. Definitely like the idea of using Notify instead, let's get this in then make that change.

**Review by oliverfoster — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by tomgreenfield — APPROVED** [review]

**Pull request merged**